### PR TITLE
feat(servicestage/app): add new resource to manage applications

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new resource to manage applications and according to the ver.3 APIs.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to manage applications.
```

## PR Checklist

* [x] Tests added/passed.

```
../coverage.sh -o servicestage -f TestAccV3Application_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/servicestage" -v -coverprofile="./huaweicloud/services/acceptance/servicestage_coverage.cov" -coverpkg="./huaweicloud/services/servicestage" -run TestAccV3Application_basic -timeout 360m -parallel 10
=== RUN   TestAccV3Application_basic
=== PAUSE TestAccV3Application_basic
=== CONT  TestAccV3Application_basic
--- PASS: TestAccV3Application_basic (107.38s)
PASS
coverage: 9.2% of statements in ./huaweicloud/services/servicestage
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/servicestage      107.436s        coverage: 9.2% of statements in ./huaweicloud/services/servicestage
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    image

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    image

### CODE REVIEW

File: huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_application_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/servicestage"
+)
+
+func getV3AppFunc(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := conf.NewServiceClient("servicestage", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating ServiceStage v2 client: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_application_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating ServiceStage client: %s", err)
```

File: huaweicloud/services/servicestage/resource_huaweicloud_servicestagev3_application.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = strings.ReplaceAll(deletePath, "{application_id}", appId)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+		OkCodes: []int{204},
```
Comment (deer-hang) [inline huaweicloud/services/servicestage/resource_huaweicloud_servicestagev3_application.go:263]:
```suggestion
		OkCodes: []int{200, 202, 204},
```

File: huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_application_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/servicestage/resource_huaweicloud_servicestagev3_application_test.go]: done.

**Pull request merged**